src/plugins/data/common/search/aggs/buckets/lib/cidr_mask.test.ts

```diff
@@ -9,56 +9,47 @@
 import { CidrMask } from './cidr_mask';
 
 describe('CidrMask', () => {
-  test('should throw errors with invalid CIDR masks', () => {
-    expect(
-      () =>
-        // @ts-ignore
-        new CidrMask()
-    ).toThrowError();
-
-    expect(() => new CidrMask('')).toThrowError();
-    expect(() => new CidrMask('hello, world')).toThrowError();
-    expect(() => new CidrMask('0.0.0.0')).toThrowError();
-    expect(() => new CidrMask('0.0.0.0/0')).toThrowError();
-    expect(() => new CidrMask('0.0.0.0/33')).toThrowError();
-    expect(() => new CidrMask('256.0.0.0/32')).toThrowError();
-    expect(() => new CidrMask('0.0.0.0/32/32')).toThrowError();
-    expect(() => new CidrMask('1.2.3/1')).toThrowError();
-    expect(() => new CidrMask('0.0.0.0/123d')).toThrowError();
+  describe('constructor', () => {
+    it.each`
+      mask
+      ${''}
+      ${'hello, world'}
+      ${'0.0.0.0'}
+      ${'0.0.0.0/33'}
+      ${'256.0.0.0/32'}
+      ${'0.0.0.0/32/32'}
+      ${'0.0.0.0/123d'}
+      ${'::1'}
+      ${'::1/129'}
+      ${'fffff::/128'}
+      ${'ffff::/128/128'}
+    `('should throw an error on $mask', ({ mask }) => {
+      expect(() => new CidrMask(mask)).toThrowError();
+    });
   });
 
-  test('should correctly grab IP address and prefix length', () => {
-    let mask = new CidrMask('0.0.0.0/1');
-    expect(mask.initialAddress.toString()).toBe('0.0.0.0');
-    expect(mask.prefixLength).toBe(1);
-
-    mask = new CidrMask('128.0.0.1/31');
-    expect(mask.initialAddress.toString()).toBe('128.0.0.1');
-    expect(mask.prefixLength).toBe(31);
+  describe('toString', () => {
+    it.each`
+      mask                        | expected
+      ${'192.168.1.1/24'}         | ${'192.168.1.1/24'}
+      ${'192.168.257/32'}         | ${'192.168.1.1/32'}
```

Comment:
Curious, what does it mean when IP component is 257 (greater than 255)?

Reply:
The IPv4 itself is a 32-bit number. When it has less than 4 parts, the missing parts can be gathered from the related bytes from the last number (the test demonstrates that case). Actually, that works oppositely, but that's to simplify the explanation. [Here](https://en.wikipedia.org/wiki/IPv4#Address_representations) is another explanation.